docs/src/main/asciidoc/secret-management.adoc

```diff
@@ -14,165 +19,188 @@ spring-cloud-azure-starter-keyvault-secrets adds Azure Key Vault as one of the S
+| *spring.cloud.azure.keyvault.secret*.property-sources[].proxy           |                                  | Proxy related properties.
+| *spring.cloud.azure.keyvault.secret*.property-sources[].retry           |                                  | Retry related properties.
+|===
+
+- Please refer to link:index.html#authorize-access-with-azure-active-directory[Authorize access with Azure AD] to make sure the link:https://docs.microsoft.com/azure/active-directory/develop/app-objects-and-service-principals#service-principal-object[security principal] has been granted the sufficient permission to access the Azure Key Vault secrets.
+- If common properties like `client`, `credential`, `profile`, `proxy`, `retry` are not configured in `spring.cloud.azure.keyvault.secret.property-sources[].xxx`, `spring.cloud.azure.xxx` will be used. Please refer to link:index.html#configuration[Configuration] to get more information about these common properties.
+- Please refer to link:appendix.html#_configuration_properties[Configuration Properties] to get more information about nested properties.
+
+==== Samples
+
+Sample project: link:https://github.com/Azure-Samples/azure-spring-boot-samples/tree/spring-cloud-azure_{project-version}/keyvault/spring-cloud-azure-starter-keyvault-secrets/property-source[property-source].
+
+
+=== Auto Configure beans for Azure Key Vault Secret Clients
+
+==== Dependency Setup
+
+[source,xml]
+----
+<dependency>
+    <groupId>com.azure.spring</groupId>
+    <artifactId>spring-cloud-azure-starter-keyvault-secrets</artifactId>
+</dependency>
+----
+
+==== Basic Usage
+
+===== Configuration Properties
 [source,yml]
 ----
 spring:
   cloud:
     azure:
       keyvault:
         secret:
-          property-source-enabled: true
-          property-sources:
-          -
-            name: key-vault-1
-            endpoint: ${ENDPOINT_1}
-            profile:
-              tenant-id: ${AZURE_TENANT_ID_1}
-            credential:
-              client-id: ${AZURE_CLIENT_ID_1}
-              client-secret: ${AZURE_CLIENT_SECRET_1}
-          -
-            name: key-vault-2
-            endpoint: ${ENDPOINT_2}
-            profile:
-              tenant-id: ${AZURE_TENANT_ID_2}
-            credential:
-              client-id: ${AZURE_CLIENT_ID_2}
-              client-secret: ${AZURE_CLIENT_SECRET_2}
-
+          endpoint: ${AZURE_KEYVAULT_ENDPOINT}
 ----
-Same as above, properties like `tenant-id`, `client-id`, `client-secret` are not necessary if authenticate by other methods.
-
-===== Java Code
 
+===== Java code
 [source,java]
 ----
+package com.azure.spring.keyvault.secrets.sample.secret.client;
+
+import com.azure.security.keyvault.secrets.SecretClient;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
```

Comment:
remove this

Reply:
take a look at spring's ref doc, they don't use the imports https://docs.spring.io/spring-cloud-stream-binder-rabbit/docs/3.0.10.RELEASE/reference/html/spring-cloud-stream-binder-rabbit.html#_putting_it_all_together